### PR TITLE
Fix keras.py

### PR DESCRIPTION
**Pull request by hunse**
Fixed keras.py to work with new Keras/tensorflow

### CODE REVIEW

**Comment by Seanny123** [comment]
I'm assuming this is a work in progress, but let me know when I can review. It would be really awesome to use the new Keras API.

File: nengo_extras/keras.py [hunk 1]
```diff
@@ -111,18 +111,29 @@ def _add_dense_layer(self, layer):
...
+
+        if layer.data_format == 'channels_first':
+            nc, _, _ = shape_in
+        elif layer.data_format == 'channels_last':
+            _, _, nc = shape_in
+
+        if K.backend() == 'tensorflow':
+            filters = np.transpose(filters, (3, 2, 0, 1))
+
+        nf, nc2, ni, nj = filters.shape
```
Comment (hunse) [inline nengo_extras/keras.py]: Should be `si` and `sj`

**Comment by hunse** [comment]
No, not WIP. There's a bug, though.

**Comment by hunse** [comment]
Ok, this is ready for review/merge.

File: examples/keras/mnist_spiking_cnn.py [hunk 2]
```diff
@@ -53,29 +63,67 @@
     kmodel = Sequential()
 
     softlif_params = dict(
-        sigma=0.002, amplitude=0.063, tau_rc=0.022, tau_ref=0.002)
-
-    kmodel.add(GaussianNoise(0.1, input_shape=(1, img_rows, img_cols)))
-    kmodel.add(Convolution2D(nb_filters, nb_conv, nb_conv, border_mode='valid'))
+        # sigma=0.02, amplitude=0.063, tau_rc=0.05, tau_ref=0.001)
```
Comment (Seanny123) [inline examples/keras/mnist_spiking_cnn.py]: Would you mind switching this to a comment suggesting other parameters to try? These commented lines just hanging around without context/explanation is a bit confusing.

Comment (Seanny123) [reply examples/keras/mnist_spiking_cnn.py]: I see now there's a lot of commented lines left over. Do you want to remove or explain them before merging?

Comment (hunse) [reply examples/keras/mnist_spiking_cnn.py]: Yes, I do :)

**Pull request merged**